packages/services/src/providers/spot-live/alchemy-rpc.provider.ts

```diff
@@ -0,0 +1,900 @@
+
+            // Convert block numbers to hex format if they're numbers
+            const fromBlockHex =
+              typeof fromBlock === "number"
+                ? `0x${fromBlock.toString(16)}`
+                : fromBlock;
+            const toBlockHex =
+              typeof toBlock === "number"
+                ? `0x${toBlock.toString(16)}`
+                : toBlock;
+
+            // Get all transfers where the wallet is involved
+            const [fromTransfers, toTransfers] = await Promise.all([
+              // Outgoing transfers (wallet is sender)
+              provider.core.getAssetTransfers({
+                fromBlock: fromBlockHex,
+                toBlock: toBlockHex,
+                fromAddress: walletAddress,
+                category: [
+                  AssetTransfersCategory.EXTERNAL, // ETH transfers
+                  AssetTransfersCategory.ERC20, // Token transfers
+                ],
+                withMetadata: true,
+                excludeZeroValue: false, // Include zero-value transfers for completeness
+                maxCount: 1000, // Alchemy's max per request
+              }),
+              // Incoming transfers (wallet is receiver)
+              provider.core.getAssetTransfers({
+                fromBlock: fromBlockHex,
+                toBlock: toBlockHex,
+                toAddress: walletAddress,
+                category: [
+                  AssetTransfersCategory.EXTERNAL,
+                  AssetTransfersCategory.ERC20,
+                ],
+                withMetadata: true,
+                excludeZeroValue: false,
+                maxCount: 1000,
+              }),
+            ]);
+
+            return {
+              transfers: [...fromTransfers.transfers, ...toTransfers.transfers],
+              pageKey: fromTransfers.pageKey || toTransfers.pageKey,
+            };
+          },
+          {
+            maxRetries: this.maxRetries,
+            initialDelay: this.retryDelayMs,
+            onRetry: ({ attempt, nextDelayMs, error }) => {
+              this.logger.warn(
+                {
+                  error: error instanceof Error ? error.message : String(error),
+                  attempt,
+                  nextDelayMs,
+                  address: maskedAddress,
+                  chain,
+                },
+                "[AlchemyRpcProvider] Retrying asset transfers request",
+              );
+            },
+          },
+        );
+      });
+
+      // Sampling for raw data storage (1% of requests)
+      const shouldSample = Math.random() < this.SAMPLING_RATE;
+
+      if (shouldSample) {
+        // Send to Sentry for monitoring
+        Sentry.captureMessage("Alchemy RPC Response Sample", {
+          level: "debug",
+          extra: {
+            response: {
+              transferCount: result.transfers.length,
+              hasPageKey: !!result.pageKey,
+              sampleTransfers: result.transfers.slice(0, 3), // First 3 transfers as sample
```

Comment:
Is it okay that the transfers have unmasked from and to's

Reply:
Ended up fixing this in a downstream PR: https://github.com/recallnet/js-recall/pull/1596